projects/developer/src/app/common/components/recipe-graph/component.ts

```diff
@@ -1180,4 +1180,17 @@ export class RecipeGraphComponent implements OnInit, OnChanges, AfterViewInit {
             this.update.next(true);
         }, 0);
     }
+
+    public hasConnection(filename, connections) {

```

Comment:
Could you specify types for `filename`,  `connections`, and the return type? Also add a docstring to describe what this method is for.

Reply:
Sure.